src/test/unit/profile-upgrading.test.js

```diff
@@ -0,0 +1,285 @@
+    const upgradedProfile7 = unserializeProfileOfArbitraryFormat(
+      serializedOldProcessedProfile7
+    );
+    compareProcessedProfiles(upgradedProfile7, afterUpgradeReference);
+  });
+  it('should upgrade version 8', function() {
+    // processed-7a to processed-8a is testing that we properly
+    // upgrade the DOMEventMarkerPayload.timeStamp field.
+    const serializedOldProcessedProfile7a = require('../fixtures/upgrades/processed-7a.json');
+    const afterUpgradeReference8a = unserializeProfileOfArbitraryFormat(
+      require('../fixtures/upgrades/processed-8a.json')
+    );
+    const upgradedProfile7a = unserializeProfileOfArbitraryFormat(
+      serializedOldProcessedProfile7a
+    );
+    compareProcessedProfiles(upgradedProfile7a, afterUpgradeReference8a);
+  });
+  it('should upgrade version 9', function() {
+    const serializedOldProcessedProfile8 = require('../fixtures/upgrades/processed-8.json');
+    const upgradedProfile8 = unserializeProfileOfArbitraryFormat(
+      serializedOldProcessedProfile8
+    );
+    compareProcessedProfiles(upgradedProfile8, afterUpgradeReference);
+  });
+  it('should upgrade version 10', function() {
+    const serializedOldProcessedProfile9 = require('../fixtures/upgrades/processed-9.json');
+    const upgradedProfile9 = unserializeProfileOfArbitraryFormat(
+      serializedOldProcessedProfile9
+    );
+    compareProcessedProfiles(upgradedProfile9, afterUpgradeReference);
+  });
+  it('should upgrade version 11', function() {
+    const serializedOldProcessedProfile11 = require('../fixtures/upgrades/processed-11.json');
+    const upgradedProfile11 = unserializeProfileOfArbitraryFormat(
+      serializedOldProcessedProfile11
+    );
+    compareProcessedProfiles(upgradedProfile11, afterUpgradeReference);
+  });
+  it('should upgrade version 12', function() {
+    const serializedOldProcessedProfile12 = require('../fixtures/upgrades/processed-12.json');
+    const upgradedProfile12 = unserializeProfileOfArbitraryFormat(
+      serializedOldProcessedProfile12
+    );
+    compareProcessedProfiles(upgradedProfile12, afterUpgradeReference);
+  });
+  it('should still process a profile of the current version with no issues', function() {
+    const serializedOldProcessedProfile13 = require('../fixtures/upgrades/processed-13.json');
+    const upgradedProfile13 = unserializeProfileOfArbitraryFormat(
+      serializedOldProcessedProfile14
+    );
+    compareProcessedProfiles(upgradedProfile13, afterUpgradeReference);
+  });
+});
+
+describe('upgrading gecko profiles', function() {
+  const afterUpgradeGeckoReference = require('../fixtures/upgrades/gecko-11.json');
+  // Uncomment this to output your next ./upgrades/gecko-X.json
+  // upgradeGeckoProfileToCurrentVersion(afterUpgradeGeckoReference);
+  // console.log(JSON.stringify(afterUpgradeGeckoReference));
+  // Then run prettier on it with the following command:
+  //   yarn run prettier --write <file name>
```

Comment:
This is definitely a pain point on working with upgrading things. It's probably beyond the scope of this PR to do anything major to fix it. I don't have great ideas. I briefly thought about using Jest's snapshot feature to try and automate things, but the problem is that you want to capture the old format, and upgrade from every single old format to the old one, to help ensure we are truly backwards compatible.

We could potentially do some magic automation, where we write the new format to file using some node script, but I think it's probably better to keep it simple. If we think of something concrete let's file a bug to make it nicer, although I'm not sure of anything at this point.

Reply:
Oh, and to be clear, the diff might make it look like this is new code, but this is copied pretty much verbatim from the original tests, but I changed the indentation and code location.

Reply:
I wanted to point it again since I had to write an upgrader once an it was really pain for me to understand and update as someone who was new to codebase. But you're right. This is not in the scope of this PR :)